### PR TITLE
JDP230502-22 - Group Controller, mapper and service + GroupServiceTests

### PR DESCRIPTION
**Pull request by annapawlas**

### CODE REVIEW

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java [hunk 1]
```diff
@@ -20,23 +20,22 @@ public class Cart {
     @Column(name = "cartId", unique = true)
     private Long id;
 
-    @OneToOne(mappedBy = "cart")
+    @OneToOne(cascade = CascadeType.ALL, mappedBy = "cart")
```
Comment (sew-k) [inline src/main/java/com/kodilla/ecommercee/domain/Cart.java]: To be consulted with other pull-request (JDP230502-19) - if merged. If not, I suggest to check if other parameters of cascade would be more suitable.

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java [hunk 2]
```diff
@@ -20,23 +20,22 @@ public class Cart {
...
     @JoinColumn(name = "userId")
     private User user;
 
-    @OneToOne(mappedBy = "cart")
+    @OneToOne(cascade = CascadeType.ALL, mappedBy = "cart")
     private Order order;
     @OneToMany(
             targetEntity =  CartProducts.class,
             fetch = FetchType.LAZY,
+            cascade = CascadeType.ALL,
```
Comment (sew-k) [inline src/main/java/com/kodilla/ecommercee/domain/Cart.java]: I suggest better not to delete those entities, when removing cart.

File: src/main/java/com/kodilla/ecommercee/service/GroupService.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    public void createGroup(Group group) {
+        groupRepository.save(group);
+    }
+
+    public List<Group> getAllGroups() {
+        return groupRepository.findAll();
+    }
+
+    public Group getGroup(final Long id) {
```
Comment (sew-k) [inline src/main/java/com/kodilla/ecommercee/service/GroupService.java:26]: I think "throws GroupNotFoundException" would be needed.

File: src/main/java/com/kodilla/ecommercee/service/GroupService.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        return groupRepository.findById(id).orElseThrow(GroupNotFoundException::new);
+    }
+
+    public void deleteGroup(final Long id) {
```
Comment (sew-k) [inline src/main/java/com/kodilla/ecommercee/service/GroupService.java:30]: I think "throws GroupNotFoundException" would be needed.

File: src/main/java/com/kodilla/ecommercee/service/GroupService.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        groupRepository.findById(id).orElseThrow(GroupNotFoundException::new);
+        groupRepository.deleteById(id);
+    }
+
+    public Group updateGroup(final Long id, String name) {
```
Comment (sew-k) [inline src/main/java/com/kodilla/ecommercee/service/GroupService.java:35]: I think "throws GroupNotFoundException" would be needed.

File: src/main/java/com/kodilla/ecommercee/domain/Cart.java (hunk 1 above)
Comment (sew-k) [reply src/main/java/com/kodilla/ecommercee/domain/Cart.java]: JDP230502-19 is merged. Here cascadeType is the same there.

**Review by sew-k — APPROVED** [review]

File: src/main/java/com/kodilla/ecommercee/service/GroupService.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+    public void createGroup(Group group) {
+        groupRepository.save(group);
+    }
+
+    public List<Group> getAllGroups() {
+        return groupRepository.findAll();
+    }
+
+    public Group getGroup(final Long id) throws GroupNotFoundException {
```
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/service/GroupService.java]: `throws GroupNotFoundException` is redundant - you make exception in lower line `orElseThrow(GroupNotFoundException::new)`

File: src/main/java/com/kodilla/ecommercee/service/GroupService.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        return groupRepository.findById(id).orElseThrow(GroupNotFoundException::new);
+    }
+
+    public void deleteGroup(final Long id) throws GroupNotFoundException {
```
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/service/GroupService.java]: `throws GroupNotFoundException` is redundant - you make exception in lower line `orElseThrow(GroupNotFoundException::new)`

File: src/main/java/com/kodilla/ecommercee/service/GroupService.java [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        groupRepository.findById(id).orElseThrow(GroupNotFoundException::new);
+        groupRepository.deleteById(id);
+    }
+
+    public Group updateGroup(final Long id, String name) throws GroupNotFoundException {
```
Comment (WorstWorm) [inline src/main/java/com/kodilla/ecommercee/service/GroupService.java]: `throws GroupNotFoundException` is redundant - you make exception in lower line `orElseThrow(GroupNotFoundException::new)`

**Review by WorstWorm — APPROVED** [review]
:thumbsup:

**Pull request merged**